[CI] XPackRestIT test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version} failing

**Build Scans:**
- [elasticsearch-periodic #8937 / release-tests](https://gradle-enterprise.elastic.co/s/f266d3cqbxdui)
- [elasticsearch-periodic #8919 / release-tests](https://gradle-enterprise.elastic.co/s/4x5r64cjwvyj4)
- [elasticsearch-pull-request #76356 / part-3](https://gradle-enterprise.elastic.co/s/phaarzheopdkk)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:yamlRestTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version}" -Dtests.seed=9CCC1B48293E2B32 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=en-DE -Dtests.timezone=Europe/Isle_of_Man -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.test.rest.XPackRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Desql%2F60_usage%2FBasic%20ESQL%20usage%20output%20(telemetry)%20non-snapshot%20version%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [esql/60_usage:149]: field [esql.features] doesn't have length [26]
Expected: <26>
     but: was <27>
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version}
- [main] 2 consecutive failures in step release-tests
- [main] 2 consecutive failures in pipeline elasticsearch-periodic
- [main] 3 failures in test test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version} (18.8% fail rate in 16 executions)
- [main] 2 failures in step release-tests (18.2% fail rate in 11 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (18.2% fail rate in 11 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -562,9 +562,6 @@ tests:
 - class: org.elasticsearch.xpack.rank.rrf.RRFRankClientYamlTestSuiteIT
   method: test {yaml=rrf/950_pinned_interaction/rrf with pinned retriever as a sub-retriever}
   issue: https://github.com/elastic/elasticsearch/issues/129845
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version}
-  issue: https://github.com/elastic/elasticsearch/issues/129888
 - class: org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapperTests
   method: testExistsQueryMinimalMapping
   issue: https://github.com/elastic/elasticsearch/issues/129911

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java
@@ -120,6 +120,13 @@ public class LogicalPlanBuilder extends ExpressionBuilder {
 
     protected LogicalPlan plan(ParseTree ctx) {
         LogicalPlan p = ParserUtils.typedParsing(this, ctx, LogicalPlan.class);
+        if (p instanceof Explain == false && p.anyMatch(logicalPlan -> logicalPlan instanceof Explain)) {
+            throw new ParsingException(source(ctx), "EXPLAIN does not support downstream commands");
+        }
+        if (p instanceof Explain explain && explain.query().anyMatch(logicalPlan -> logicalPlan instanceof Explain)) {
+            // TODO this one is never reached because the Parser fails to understand multiple round brackets
+            throw new ParsingException(source(ctx), "EXPLAIN cannot be used inside another EXPLAIN command");
+        }
         var errors = this.context.params().parsingErrors();
         if (errors.hasNext() == false) {
             return p;

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java
@@ -190,15 +190,10 @@ public class EsqlSession {
         assert executionInfo != null : "Null EsqlExecutionInfo";
         LOGGER.debug("ESQL query:\n{}", request.query());
         LogicalPlan parsed = parse(request.query(), request.params());
-        Explain explain = findExplain(parsed);
-        if (explain != null) {
+        if (parsed instanceof Explain explain) {
             explainMode = true;
-            if (explain == parsed) {
-                parsed = explain.query();
-                parsedPlanString = parsed.toString();
-            } else {
-                throw new VerificationException("EXPLAIN does not support downstream commands");
-            }
+            parsed = explain.query();
+            parsedPlanString = parsed.toString();
         }
         analyzedPlan(parsed, executionInfo, request.filter(), new EsqlCCSUtils.CssPartialErrorsActionListener(executionInfo, listener) {
             @Override
@@ -209,19 +204,6 @@ public class EsqlSession {
                 );
             }
         });
-    }
-
-    private Explain findExplain(LogicalPlan parsed) {
-        if (parsed instanceof Explain e) {
-            return e;
-        }
-        for (LogicalPlan child : parsed.children()) {
-            Explain result = findExplain(child);
-            if (result != null) {
-                return result;
-            }
-        }
-        return null;
     }
 
     /**

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java
@@ -3500,7 +3500,10 @@ public class StatementParserTests extends AbstractStatementParserTests {
         assumeTrue("Requires EXPLAIN capability", EsqlCapabilities.Cap.EXPLAIN.isEnabled());
         // TODO this one is incorrect
         expectError("explain ( from test ) | limit 1", "line 1:23: mismatched input '|' expecting {'|', ',', ')', 'metadata'}");
-        expectError("explain (row x=\"Elastic\" | eval y=concat(x,to_upper(\"search\"))) | mv_expand y", "line 1:1: EXPLAIN does not support downstream commands");
+        expectError(
+            "explain (row x=\"Elastic\" | eval y=concat(x,to_upper(\"search\"))) | mv_expand y",
+            "line 1:1: EXPLAIN does not support downstream commands"
+        );
     }
 
     public void testRerankDefaultInferenceIdAndScoreAttribute() {

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java
@@ -3496,6 +3496,13 @@ public class StatementParserTests extends AbstractStatementParserTests {
         }
     }
 
+    public void testExplainErrors() {
+        assumeTrue("Requires EXPLAIN capability", EsqlCapabilities.Cap.EXPLAIN.isEnabled());
+        // TODO this one is incorrect
+        expectError("explain ( from test ) | limit 1", "line 1:23: mismatched input '|' expecting {'|', ',', ')', 'metadata'}");
+        expectError("explain (row x=\"Elastic\" | eval y=concat(x,to_upper(\"search\"))) | mv_expand y", "line 1:1: EXPLAIN does not support downstream commands");
+    }
+
     public void testRerankDefaultInferenceIdAndScoreAttribute() {
         assumeTrue("RERANK requires corresponding capability", EsqlCapabilities.Cap.RERANK.isEnabled());
 

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/220_explain.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/220_explain.yml
@@ -97,5 +97,5 @@ explainDownstream:
           query: 'EXPLAIN (row a = 1) | eval b = 2'
       catch: "bad_request"
 
-  - match: { error.type: "verification_exception" }
+  - match: { error.type: "parsing_exception" }
   - contains: { error.reason: "EXPLAIN does not support downstream commands" }

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml
@@ -46,7 +46,7 @@ setup:
   - do: {xpack.usage: {}}
   - match: { esql.available: true }
   - match: { esql.enabled: true }
-  - length: { esql.features: 26 }
+  - length: { esql.features: 27 }
   - set: {esql.features.dissect: dissect_counter}
   - set: {esql.features.drop: drop_counter}
   - set: {esql.features.eval: eval_counter}

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml
@@ -46,7 +46,7 @@ setup:
   - do: {xpack.usage: {}}
   - match: { esql.available: true }
   - match: { esql.enabled: true }
-  - length: { esql.features: 27 }
+  - length: { esql.features: 26 }
   - set: {esql.features.dissect: dissect_counter}
   - set: {esql.features.drop: drop_counter}
   - set: {esql.features.eval: eval_counter}

